chat/src/main/java/com/github/twitch4j/chat/TwitchChatBuilder.java

```diff
@@ -195,6 +196,21 @@
     @With
     private boolean enableMembershipEvents = true;
 
+    /**
+     * Whether join failures should result in removal from current channels.
+     *
+     * @see ChannelRemovedPostJoinFailureEvent
+     */
+    @With
+    private boolean removeChannelOnJoinFailure = false;
+
+    /**
+     * The maximum number of retries to make for joining each channel, with exponential backoff.
+     * Set to zero or a negative value to disable this feature.
+     */
+    @With
+    private int maxJoinRetries = 3;
```

Comment:
this allows 1 (the initial attempt) + 3 (the retries) = 4 join attempts before stopping (until a reconnect, assuming removal isn't enabled) - let me know if you'd prefer a different value

Reply:
I think a higher default would be better to give it a little more time if there are any weird temporary issues on either twitch or client side. I'd suggest `10`, but we can still tweak the default later.

Reply:
how about 1+7=8, so the maximum wait on default settings is around 10 mins?